File: partiql-ast/src/main/java/org/partiql/ast/v1/TableDefinition.java
package org.partiql.ast.v1;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

/**
 * TODO docs, equals, hashcode
 */
public class TableDefinition extends AstNode {
    @NotNull
    public List<Column> columns;

    public TableDefinition(@NotNull List<Column> columns) {
        this.columns = columns;
    }

    @NotNull
    @Override
    public Collection<AstNode> children() {
        return new ArrayList<>(columns);
    }

    @Override
    public <R, C> R accept(@NotNull AstVisitor<R, C> visitor, C ctx) {
        return visitor.visitTableDefinition(this, ctx);
    }
}

File: partiql-ast/src/main/kotlin/org/partiql/ast/v1/TableDefinition.kt
package org.partiql.ast.v1

/**
 * TODO docs, equals, hashcode
 */
public class TableDefinition(
    @JvmField
    public var columns: List<Column>,
) : AstNode() {
    public override fun children(): Collection<AstNode> {
        val kids = mutableListOf<AstNode?>()
        kids.addAll(columns)
        return kids.filterNotNull()
    }

    public override fun <R, C> accept(visitor: AstVisitor<R, C>, ctx: C): R =
        visitor.visitTableDefinition(this, ctx)
}
